orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala

```diff
@@ -129,11 +172,11 @@ case class Ec2Resource(name: String, spec: Ec2Resource.Spec) extends ResourceIO
           state.head match {
             case InstanceStateName.PENDING => Right(Status.Activating)
             case InstanceStateName.RUNNING =>
-              isSsmVisible(Json.toJson(Ec2Resource.InstSpec(ec2InstanceId))) match {
+              isSsmVisible(ec2InstanceId) match {
                 case true => Right(Status.Running)
                 case _ =>
                   logger.info(s"ec2InstanceId $ec2InstanceId isSsmVisible=false")
-                  Right(Status.Activating)
+                  describeEc2Status(ec2InstanceId)
```

Comment:
so if I understand correctly, first it's instance state, then there's also instance status? and we need both to determine if a instance is up and running in a healthy state?

Reply:
yes, that's my findings from testing.  Most of the time, `instanceState` is good for the newly started ec2 instances.  On rare occurrences, the `instanceStatus` is impaired.  The sequence also matters, have to `instanceState` `running` first, before `instanceStatus` OK, INITIALIZING, IMPAIRED would be returned.  Calling `describeInstanceStatus` before the `instanceState` `running` returned empty list for the matching instanceId.